### PR TITLE
FEAT: Now sends exception to data service and saves to new table in db

### PR DESCRIPTION
**Pull request by SamRobinson75684**
## Description

sends exception data to data service and saves data to database in new table

## Type of changes

- [x] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: tests/ScreeningValidationServiceTests/FileValidation/FileValidationTests.cs [hunk 1]
```diff
@@ -8,23 +8,33 @@ namespace NHS.CohortManager.Tests.ScreeningValidationService;
...
+using Model;
+using Common;
 
 [TestClass]
 public class FileValidationTests
 {
     private readonly Mock<ILogger<FileValidation>> _logger = new();
     private readonly Mock<FunctionContext> _context = new();
+    private readonly Mock<ICallFunction> _callFunction = new();
+    private readonly Mock<HttpWebResponse> webResponse = new();
```
Comment (will-larkin-nhs) [inline tests/ScreeningValidationServiceTests/FileValidation/FileValidationTests.cs:20]: should be `_webResponse`

**Review by will-larkin-nhs — CHANGES_REQUESTED** [review]

File: application/CohortManager/src/Functions/screeningDataServices/CreateValidationException/CreateValidationException.cs
Comment (alex-clayton-1) [inline application/CohortManager/src/Functions/screeningDataServices/CreateValidationException/CreateValidationException.cs:1]: The LookupValidation and StaticValidation functions call this function and pass a ValidationException, these changes will break those functions.

I think we either need a separate function for creating file validation exceptions or a db table capable of holding both types of exception.

Comment (SamRobinson75684) [reply application/CohortManager/src/Functions/screeningDataServices/CreateValidationException/CreateValidationException.cs:1]: okay so I think we can change it back for now as body of this request along with what is going to be sent is changing for everything (hence not actually writing to be db now)

**Review by alex-clayton-1 — DISMISSED** [review]

**Review by will-larkin-nhs — DISMISSED** [review]

File: application/CohortManager/src/Functions/Shared/Model/FileValidationRequestBody.cs
Comment (alex-clayton-1) [inline application/CohortManager/src/Functions/Shared/Model/FileValidationRequestBody.cs:1]: Is this. file still needed?

Comment (SamRobinson75684) [reply application/CohortManager/src/Functions/Shared/Model/FileValidationRequestBody.cs:1]: it's not yeah, good spot. I have removed this now

**Review by alex-clayton-1 — APPROVED** [review]

**Pull request merged**